packages/ui/src/lib/DeployArgumentParser.ts

```diff
@@ -0,0 +1,656 @@
+    argValueInJson: any,
+    uRefAccessRight?: Key.URef.AccessRightsMap[keyof Key.URef.AccessRightsMap]
+  ) {
+    let clValueInstance;
+    switch (firstType) {
+      case 'Bytes':
+        clValueInstance = Args.Instances.bytes(decodeBase16(argValueInJson));
+        break;
+      case 'Bytes (Fixed Length)':
+        clValueInstance = Args.Instances.bytesFixedLength(
+          decodeBase16(argValueInJson)
+        );
+        break;
+      case CLType.Simple.BOOL:
+        clValueInstance = Args.Instances.bool(argValueInJson);
+        break;
+      case CLType.Simple.I32:
+        clValueInstance = Args.Instances.i32(argValueInJson);
+        break;
+      case CLType.Simple.I64:
+        clValueInstance = Args.Instances.i64(argValueInJson);
+        break;
+      case CLType.Simple.U8:
+        clValueInstance = Args.Instances.u8(argValueInJson);
+        break;
+      case CLType.Simple.U32:
+        clValueInstance = Args.Instances.u32(argValueInJson);
+        break;
+      case CLType.Simple.U64:
+        clValueInstance = Args.Instances.u64(argValueInJson);
+        break;
+      case CLType.Simple.U128:
+        clValueInstance = Args.Instances.u128(argValueInJson);
+        break;
+      case CLType.Simple.U256:
+        clValueInstance = Args.Instances.u256(argValueInJson);
+        break;
+      case CLType.Simple.U512:
+        clValueInstance = Args.Instances.u512(argValueInJson);
+        break;
+      case CLType.Simple.UNIT:
+        clValueInstance = Args.Instances.unit();
+        break;
+      case CLType.Simple.STRING:
+        clValueInstance = Args.Instances.string(argValueInJson);
+        break;
+      case CLType.Simple.KEY:
+      case CLType.Simple.UREF:
+        clValueInstance = DeployArgumentParser.buildKeyOrUrefInstance(
+          firstType,
+          secondType,
+          argValueInJson,
+          uRefAccessRight
+        );
+        break;
+    }
+    return clValueInstance;
+  }
+
+  public static buildArgument(arg: FormState<DeployArgument>) {
+    const argValueStr: string = arg.$.value.value;
+    const type = arg.$.type.$;
+    let clValueInstance: CLValueInstance;
+    const argValueStrInJson = JSON.parse(argValueStr);
+    switch (type) {
+      case 'Tuple':
+        clValueInstance = DeployArgumentParser.buildTupleTypeArg(
+          arg.$.tupleInnerDeployArgs.$,
+          argValueStrInJson
+        );
+        break;
+      case 'Map':
+        clValueInstance = DeployArgumentParser.buildMapTypeArg(
+          arg.$.mapInnerDeployArgs.$,
+          argValueStrInJson
+        );
+        break;
+      case 'List':
+        clValueInstance = DeployArgumentParser.buildListTypeArg(
+          arg.$.listInnerDeployArgs.$,
+          argValueStrInJson,
+          false
+        );
+        break;
+      case 'FixedList':
+        clValueInstance = this.buildListTypeArg(
+          arg.$.listInnerDeployArgs.$,
+          argValueStrInJson,
+          true
+        );
+        break;
+      case 0:
```

Comment:
What those numbers stand for?

Reply:
They are from `CLType.Simple`, represent simple types like u8, u16, string and so on.